e2e/capl-cluster-flavors/kubeadm-full-capl-cluster/assert-konnectivity-resources.yaml

```diff
@@ -0,0 +1,20 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: konnectivity-agent
+  namespace: kube-system
+status:
+  availableReplicas: 3
+  readyReplicas: 3
```

Comment:
Why 3 replicas?

Reply:
The helm chart proxy deploys 3 as a standard. So by default it will be 3.

Reply:
Hmm you can set the agent value to 1 by setting that value. You may want to fix the template for this. 

Helm chart values: https://github.com/linode/konnectivity/blob/09e721a56af589cf1ec2d9c692c2f73f48ba4bc5/chart/values.yaml#L25

Reply:
You can add something like:
```
diff --git a/templates/addons/konnectivity/konnectivity.yaml b/templates/addons/konnectivity/konnectivity.yaml
index d0bbfd0..659028c 100644
--- a/templates/addons/konnectivity/konnectivity.yaml
+++ b/templates/addons/konnectivity/konnectivity.yaml
@@ -18,3 +18,4 @@ spec:
     proxyServerHost: {{ .InfraCluster.spec.controlPlaneEndpoint.host }}
     proxyServerPort: ${KONNECTIVITY_PORT:=8132}
     serverCount: ${CONTROL_PLANE_MACHINE_COUNT}
+    agentReplicas: ${KONNECTIVITY_AGENT_REPLICAS:=3}
```
and then set `KONNECTIVITY_AGENT_REPLICAS` to 1 when provisioning the cluster.

Reply:
Say for example we generate a cluster using this command 

clusterctl generate cluster kb-full-test --flavor kubeadm-full --infrastructure local-linode:v0.0.0 --control-plane-machine-count=1 --worker-machine-count=1 > xyz.yaml

Can we do;

agentReplicas: ${KONNECTIVITY_AGENT_REPLICAS:=WORKER_MACHINE_COUNT}